mobile/src/main/java/rs/readahead/washington/mobile/views/activity/CollectFormEntryActivity.kt

```diff
@@ -0,0 +1,1007 @@
+    private var progressDialog: ProgressDialog? = null
+    private var deleteEnabled = false
+    private var draftAutoSaved = false
+    private var micFragment: MicFragment? = null
+    private lateinit var binding: ActivityCollectFormEntryBinding
+
+    //private val model: SharedFormsViewModel by viewModels()
+    private val viewModel: SubmitFormsViewModel by viewModels()
+
+    public override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivityCollectFormEntryBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+        currentScreenView = null
+        //sectionIndex = 0;
+        setSupportActionBar(binding.toolbar)
+        val actionBar = supportActionBar
+        actionBar?.setDisplayHomeAsUpEnabled(true)
+        upNavigationIcon = binding.toolbar.navigationIcon
+        setToolbarIcon()
+        initForm()
+        startPresenter()
+        initObservers()
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+            binding.appbar.outlineProvider = null
+        } else {
+            binding.appbar.bringToFront()
+        }
+        binding.prevSection.setOnClickListener { v -> showPrevScreen() }
+        binding.nextSection.setOnClickListener { v -> showNextScreen() }
+        binding.submitButton.setOnClickListener { v ->
+            viewModel.submitActiveFormInstance(formTitle + " " + Util.getDateTimeString())
+            hideToolbarIcon()
+            hideSubmitButtons()
+            showFormCancelButton()
+        }
+        binding.cancelButton.setOnClickListener { v ->
+            if (userStopPresenterSubmission()) {
+                hideFormCancelButton()
+            }
+        }
+        endView = CollectFormEndView(this, R.string.Uwazi_Submitted_Entity_Header_Title)
+
+        /*disposables.wire(
+            FormAttachmentsUpdatedEvent::class.java,
+            object : EventObserver<FormAttachmentsUpdatedEvent?>() {
+                override fun onNext(event: FormAttachmentsUpdatedEvent) {
+                    formAttachmentsChanged()
+                }
+            })
+         disposables.wire(
+             LocationPermissionRequiredEvent::class.java,
+             object : EventObserver<LocationPermissionRequiredEvent?>() {
+                 override fun onNext(event: LocationPermissionRequiredEvent) {
+                     //CollectFormEntryActivityPermissionsDispatcher.startPermissionProcessWithPermissionCheck(CollectFormEntryActivity.this);
+                 }
+             })
+         disposables.wire(
+             GPSProviderRequiredEvent::class.java,
+             object : EventObserver<GPSProviderRequiredEvent?>() {
+                 override fun onNext(event: GPSProviderRequiredEvent) {
+                     // CollectFormEntryActivityPermissionsDispatcher.startPermissionProcessWithPermissionCheck(CollectFormEntryActivity.this);
+                 }
+             })*/
+        disposables.wire(
```

Comment:
maybe we could call some method here instead of putting code in the onCreate and also remove commented code